### PR TITLE
Update minimum CMake version to 3.22

### PR DESCRIPTION
**Pull request by amzn-changml**
Signed-off-by: Mike Chang <changml@amazon.com>

## What does this PR do?

Bumps the minimum version of CMake to 3.22. This enables some features that are not fully supported in our current 3.20 minimum, including:

- VS2022 generation (Introduced in 3.21: https://cmake.org/cmake/help/latest/release/3.21.html)
- CMake presets (Introduced in 3.19, but was fully supported in 3.21: https://cmake.org/cmake/help/latest/release/3.21.html)

We also have the option of doing a minor bump to 3.21 just to get the above features in, but would rather err towards pulling as close to latest as possible for bugfixes. In addition Ubuntu 22.04 will default to CMake 3.22 from Apt, as will Windows and Mac package managers (Chocolatey and Homebrew respectively).

## How was this PR tested?

Run in a sandbox Jenkins CI with 3.24 installed.

### CODE REVIEW

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by spham-amzn — APPROVED** [review]
Make sure to do a docs PR as well

**Review by brianherrera — APPROVED** [review]
Also make sure to update README.md

**Review by hultonha — APPROVED** [review]

File: AutomatedTesting/CMakeLists.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #
 
 if(NOT PROJECT_NAME)
-    cmake_minimum_required(VERSION 3.20)
+    cmake_minimum_required(VERSION 3.24)
```
Comment (spham-amzn) [inline AutomatedTesting/CMakeLists.txt:10]: I just noticed that Ubuntu 22.04 defaults to cmake 3.22, not 3.24. ( https://packages.ubuntu.com/jammy/cmake ). Is it possible to downgrade this to cmake 3.22 as the minimum instead?

**Comment by spham-amzn** [comment]
I just noticed that Ubuntu 22.04 defaults to cmake 3.22, not 3.24. ( https://packages.ubuntu.com/jammy/cmake ). Is it possible to downgrade this to cmake 3.22 as the minimum instead?

**Comment by Darkinggq** [comment]
Fedora 36 cmake version (Fedora 37 beta cmake 3.24)

```
~]$ cmake -version
cmake version 3.22.2

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```
image
image
image

this is taken from the site https://pkgs.org/search/?q=cmake other distributions

better minimum 3.22 and not 3.24

cmake 3.24 too early

**Comment by Darkinggq** [comment]
today updated cmake 3.22 to 3.24 for fedora 36 (Red Hat), ubuntu 22.04 lts with old 3.22

image

if we want to update cmake we will have to delete and repo added or install snap cmake to the latest version this is for ubuntu 20.04/22.04 https://snapcraft.io/cmake

File: AutomatedTesting/CMakeLists.txt (hunk 1 above)
Comment (amzn-changml) [reply AutomatedTesting/CMakeLists.txt:10]: Yes, that should be ok. I'll change this and make a new announcement for that on Monday.

**Comment by amzn-changml** [comment]
> 



> today updated cmake 3.22 to 3.24 for fedora 36 (Red Hat), ubuntu 22.04 lts with old 3.22
> 
> image
> 
> if we want to update cmake we will have to delete and repo added or install snap cmake to the latest version this is for ubuntu 20.04/22.04 https://snapcraft.io/cmake

Thanks for that data point. I'll go ahead and downgrade this to 3.22. Seems like we'll have lots of friction for Linux distros. The features we're looking to use are still compatible with that version.

**Review by hultonha — APPROVED** [review]

**Review by spham-amzn — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**